### PR TITLE
HL Python SDK: Improve reference types docs

### LINKED ISSUE #7171: HL Python SDK: Improve Documentation of reference types
Since all of the reference types (Branch, Tag) inherit from Reference their class documentation looks like this:
image

Explicitly add __init__ for these classes and provide better naming to parameters so that documentation will be more clear

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #7171

### Before:

image

### After:

image

### CODE REVIEW

**Comment by N-o-Z** [comment]
@talSofer  FYI

File: clients/python-wrapper/docs/conf.py [hunk 1]
```diff
@@ -50,7 +50,7 @@
 
 def skip(app, what, name, obj, would_skip, options):
     if name in ('__init__',):
-        return False
+        return True
     return would_skip
```
Comment (nopcoder) [inline clients/python-wrapper/docs/conf.py:54]: we can just `return name in ('__init__'):` (no ',' too)

**Review by nopcoder — APPROVED** [review]
LGTM - Glad we make repo_id -> repository_id soon as existing code can use named parameter to call these methods.
Small commit about the `skip` func.

File: clients/python-wrapper/docs/conf.py (hunk 1 above)
Comment (N-o-Z) [reply clients/python-wrapper/docs/conf.py:54]: Removing this altogether since it's not necessary

**Pull request merged**